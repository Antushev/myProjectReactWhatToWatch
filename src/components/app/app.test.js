import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';

const films = [
  `Хоббит: Пустошь Смауга`,
  `Интерстеллар`,
  `Спасти рядоого Райана`,
  `Власталин колец: Возвращение короля`,
  `Хоббит: Нежданное путешествие`
];

const film = {
  name: `Интерстеллар`,
  genre: `Adventure`,
  date: 2014
};

describe(`AppComponent`, () => {
  it(`AppComponentSnapshot`, () => {
    const tree = renderer
      .create(
          <App
            films={films}
            filmName={film.name}
            genre={film.genre}
            date={film.date}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
