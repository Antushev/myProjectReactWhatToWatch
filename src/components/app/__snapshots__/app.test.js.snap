// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent AppComponentSnapshot authorizationStatus AUTH 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Идёт загрузка
      </h1>
    </div>
  </section>
</div>
`;

exports[`AppComponent AppComponentSnapshot isLoading false isError false 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Кажется, сервер недоступен...
      </h1>
    </div>
  </section>
</div>
`;

exports[`AppComponent AppComponentSnapshot isLoading false isError true 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Кажется, сервер недоступен...
      </h1>
    </div>
  </section>
</div>
`;

exports[`AppComponent AppComponentSnapshot isLoading true isError false authorizationStatus NO_AUTH 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Идёт загрузка
      </h1>
    </div>
  </section>
</div>
`;

exports[`AppComponent AppComponentSnapshot isLoading true isError true 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <div
      className="catalog__movies-list"
    >
      <h1>
        Идёт загрузка
      </h1>
    </div>
  </section>
</div>
`;
